assets/src/js/search.js

```diff
@@ -84,42 +91,101 @@ export const setupSearch = function($) {
+          'search_query':           $( '#search_input' ).val().trim(),
+          'total_posts':            total_posts,
+          'archive_and_live_loads': archive_and_live_loads,
+          'paged':                  next_page,
+          'query-string':           decodeURIComponent( location.search ).substr( 1 ), // Ignore the ? in the search url (first char).
+          'last_archive_seen':      lastArchiveResult ? lastArchiveResult.attr( 'href' ) : '',
         },
         dataType: 'html'
       }).done(function ( response ) {
         // Append the response at the bottom of the results and then show it.
         $( '.multiple-search-result .list-unstyled' ).append( response );
-        $( '.row-hidden:last' ).removeClass( 'row-hidden' ).show( 'fast' );
-        if (posts_per_load * next_page > total_posts) {
-          $load_more_button.hide();
+        const lastResultsWrapper = $( '.row-hidden:last' );
+        let lastResults = lastResultsWrapper.find( '.search-result-list-item' );
+        numberLastResults = lastResults.length;
+
+        lastResultsWrapper.removeClass( 'row-hidden' ).show( 'fast' );
+
+        if ( numberLastResults < posts_per_load ) {
+          hideLoadMore();
+        }
+
+        let loads_with_only_live_posts = ( next_page - 1 ) - archive_and_live_loads - only_archived_loads;
+        let loaded_posts               = ( archive_and_live_loads * ( posts_per_load - archives_per_load ) ) + ( loads_with_only_live_posts * posts_per_load );
+        let loaded_more_live_posts     = total_posts > loaded_posts;
+
+        if ( load_archive && loaded_more_live_posts ) {
+          archive_and_live_loads++;
+        } else if ( load_archive ) {
+          only_archived_loads++;
         }
+
+        checkShowLoadArchive( next_page );
+
       }).fail(function ( jqXHR, textStatus, errorThrown ) {
         console.log(errorThrown); //eslint-disable-line no-console
       });
     } else {
       const $row = $( '.row-hidden', $load_more_button.closest( '.container' ) );
 
       if ( 1 === $row.length ) {
-        $load_more_button.closest( '.load-more-button-div' ).hide( 'fast' );
+        hideLoadMore();
       }
       $row.first().show( 'fast' ).removeClass( 'row-hidden' );
     }
   });
 
+  $('.toggle-label').off( 'click' ).on( 'click', function() {
+    const shouldIncludeArchive = $( '.show-archive-toggle' )[0].checked;
+
+    load_archive = !shouldIncludeArchive;
+    $( '.toggle-label-include' ).toggle(shouldIncludeArchive);
+    $( '.toggle-label-exclude' ).toggle(!shouldIncludeArchive);
+  });
+
+  const hideLoadMore = () => {
+    $( '.load-more-button-div' ).hide( 'fast' );
+  };
+
+  const checkShowLoadArchive = ( current_page = 1 ) => {
+
+    if ( show_archive_button ) {
+      return;
+    }
+
+    const page_archive_available_after = $load_more_button.data('page_archive_available_after');
+
+    if ( current_page >= page_archive_available_after ) {
+      $( '.toggle-wrapper' ).show();
```

Comment:
You can assign `current_page >= page_archive_available_after` to a variable and use `toggle()` here as well so you can avoid duplication of the selector